server/src-lib/Hasura/GraphQL/Execute/LiveQuery/State.hs

```diff
@@ -142,5 +147,6 @@ removeLiveQuery lqState (LiveQueryId handlerId cohortId sinkId) = do
       if handlerIsEmpty
         then do
           STMMap.delete handlerId lqMap
+          -- TODO Nothing here would seem to imply a bug
```

Comment:
I'm assuming your question is why this isn't: `Just . _pThread <$> STM.takeTMVar ioState`? Honestly, I don't remember why `tryReadTMVar` is being used here, but looking at [this](https://github.com/hasura/graphql-engine/pull/3860/files#diff-5696cf6bf0d51987a7405e83f24d4a58R88) comment in `addLiveQuery`, this does look like a bug.